static/scripts/view.js

```diff
@@ -789,18 +789,160 @@ function checkFormValidity(event) {
+        for (i in history) {
+          var historyItem = history[i];
+          var historyItemDate = new Date(historyItem['Date']);
+          if (historyItem['Field'] != "File Name" &&
+              historyItem['Field'] != "Note Title" &&
+              historyItemDate > lastModDate) {
+                hasHistoryOverwriteableChanges = true;
+                break;
+          }
+        }
+      }
+
+      // Inform the user of changes that will be overwritten.
+      if (hasHistoryOverwriteableChanges) {
+        msg += "Your bug view is from " + lastModDate.toLocaleString();
+        msg += ", and the bug now contains changes from " + lastModHistoryDate.toLocaleString() + ".\n\n";
+        msg += "The following set of changes to this bug will likely be overwritten:\n\n";
+        for (i in history) {
+          var historyItem = history[i];
+          var historyItemDate = new Date(historyItem['Date']);
+          if (historyItem['Field'] == "File Name" ||
+              historyItem['Field'] == "Note Title") {
+                // Do not tell the user that their note or attachment
+                // may be overwritten. It won't be!
+                continue;
+          }
+          if (historyItemDate > lastModDate) {
+            if (historyItem['Operation'] == 'Delete') {
+              msg += "    Deleted '" + historyItem['Field'] + "', which was set to '" + historyItem['OldValue'] + "'\n";
+            } else if (historyItem['Operation'] == 'New Record') {
+              if (historyItem['NewValue'] == "") {
+                msg += "    Created " + historyItem['Field'] + "\n";
+              } else {
+                msg += "    Created '" + historyItem['Field'] + "' as '" + historyItem['NewValue'] + "'\n";
+              }
+            } else if (historyItem['Operation'] == 'Modify') {
+              if (historyItem['OldValue'] == "" && historyItem['NewValue'] == "") {
+                msg += "    Modified '" + historyItem['Field'] + "'\n";
+              } else if (historyItem['OldValue'] == "") {
+                msg += "    Set '" + historyItem['Field'] + "' as '" + historyItem['NewValue'] + "'\n";
+              } else {
+                msg += "    Set '" + historyItem['Field'] + "' from '" + historyItem['OldValue'] + "' to '" + historyItem['NewValue'] + "'\n";
+              }
+            } else {
+              msg += "    Performed '" + historyItem['Operation'] + "'\n";
+            }
+          }
+        }
+      }
+    } else {
+      // Failed to get the bug history.
+      // Let user still attempt submission.
+      msg += "Unable to retrieve bug history.\n";
+      msg += "Cannot confirm if past changes will be overwritten.\n";
+    }
+
+    if (msg != "") {
+      msg += "\nClick OK to submit new changes despite the above warnings.";
+      msg += "\nOtherwise, press cancel and hit refresh to update your bug view.";
+      submitChanges = confirm(msg);
+    }
+
+    if (submitChanges) {
```

Comment:
This if statement and the few lines below it are duplicated in each of the next few functions, possibly worth splitting into a separate function to avoid the duplication?

Reply:
OK